Yoast/wordpress-seo
# P3 155 flush permalinks on home url change

## Context


* We don't detect yet when the home url option is updated. We should hook into updates of the home url option. On a change, we should clear all permalinks and ask for a reindex.

## Summary


This PR can be summarized in the following changelog entry:

* Flush all permalink in indexables when the the home url option has been changed

## Relevant technical choices:

* 

## Test instructions

This PR can be tested by following these steps:

* checkout and build `trunk` branch
* make sure you have created the indexables by clicking on `Start processing and speed up your site now` (otherwise you can visit SEO → Tools)
* inspect the `wp_yoast_indexables` table in the DB to see it's populated and posts have permalinks
* visit Settings → General, and change "Site URL"
* inspect the `wp_yoast_indexables` table in the DB to see that only the the permalink in the `home-page` indexable has changed, and all the other ones still show the old home URL. This is the current behaviour that we want to change.
* visit Settings → General, and change back "Site URL" to the original value
* checkout and build the PR branch branch
* run `composer test` and see the tests are passing
* visit Settings → General, and change "Site URL"
* inspect the `wp_yoast_indexables` table in the DB to see all the permalinks have been flushed (except maybe the ones for non public posts like the ACF ones, or created by disabled plugins)
* visit the dashboard to see a notification which says:

> Because of a change in your home URL option, some of your SEO data need to be reprocessed.
We estimate this will take less than a minute.
Start processing and speed up your site now

(depending on your environment, if you try to build the indexables now the proces may be failing because the new home URL is not correctly configured in the server: but it's outside the scope of this test. You can switch bach to the correct home url and rebuild the indexables, then inspect the DB table to see they're ok now so the PR has not broken the building stage)


## UI changes

## Documentation

## Quality assurance


Fixes #
